### PR TITLE
Add a dependency on AbstractFFTs

### LINKED ISSUE #2: split package
The point of the dft.jl stuff is that you can plug in non-FFTW implementations (e.g. a pure-Julia FFT, FFTPACK, whatever) and it will still work with the same interface.   So, it really needs to be split into a separate package.

My suggestion would be that there be a separate package (maybe "AbstractFFTs"?) that contains dft.jl.  Then any package that provides FFT functionality would use this package.

Similarly the dsp stuff should probably be in a separate package.

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #2

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using FFTW
 using Base.Test
 
+importall AbstractFFTs
 importall FFTW
```
Comment (stevengj) [inline test/runtests.jl]: Why do the tests need `importall` and not just `using`?

Comment (ararslan) [inline test/runtests.jl]: To make sure these names get picked up rather than the ones from Base

**Comment by ararslan** [comment]
Wooooooooo tests are passing now! Want to take another look, @stevengj?

**Comment by stevengj** [comment]
Lgtm

**Comment by ararslan** [comment]
The macOS errors don't seem related to the package, but they are strange. I wonder if there's something I need to do differently with the Travis setup for Homebrew to work?

**Pull request merged**